src/OrchardCore.Modules/OrchardCore.Layers/Indexes/LayerMetadataIndex.cs

```diff
@@ -17,16 +17,22 @@ public override void Describe(DescribeContext<ContentItem> context)
                 .When(contentItem => contentItem.Has<LayerMetadata>())
                 .Map(contentItem =>
                 {
+                    // Remove index records of soft deleted items.
+                    if (!contentItem.Published && !contentItem.Latest)
+                    {
+                        return null;
+                    }
```

Comment:
as below, we rely on this to have a record. crucial. or items disappear.

Reply:
I think you mean as above as they are contained items, widgets in a layer, right?

If so same question

Reply:
Okay, idem i will revert this one